### ISSUE TITLE
4.14 release check list

### ISSUE THREAD

**Issue by lrineau** [CGAL/cgal#3361]
# Before the beta release
- [x] Update of the Supported platforms.
- [x] Update the `public_release_name`
- [x] Update the file `Installation/include/CGAL/version.h` (for header-only)
- [x] change the crontab to add `--beta 1` to the command line
- [x] Diff of the change log of 4.14 with 4.13 to check only 4.14 was modified.
- [x] Rewrap the file `CHANGES.md`
- [ ] Reorder the file `CHANGES.md`
- [x] Update `Documentation/doc/resources/1.8.13/menu_version.js` and the other two copies
- [x] Prepare the announcement mail of the beta release
- [x] Prepare the modification of the web site, in upcoming
- [x] open the milestone for the release
- [x] Update `menu_version.js` in the documentation.

# After the beta release
- [x] Update the `public_release_name`
- [x] change the crontab to add `--beta 1` to the command line
- [x] Update the file `Installation/include/CGAL/version.h` (for header-only)
- [x] clone the milestone
- [x] PR for brew package
- [x] PR for vcpkg
- [x] RPM for Fedora
- [x] Update `/wiki/Releases`
- [x] Update https://www.cgal.org/releases.html and change the post flags (latest-beta-release)
- [ ] Update `.htaccess` with the data from [here](https://gist.github.com/sloriot/4ccb76b96a3558c8177c3eda914e67a0) + test for example AABB-tree + Triangulation_2 (not sure about the last rule that seems a bit generic).
- [x] Upload the pre-compiled demos

# Before the official release
- [x] Update the `public_release_name`
- [x] Update the file `Installation/include/CGAL/version.h` (for header-only)
- [x] change the crontab to remove `--beta 1` from the command line
- [x] Update `menu_version.js` in the documentation.

# After the official release
- [x] Update `/home/lrineau/CGAL/create_internal_release` on cgal.gf.com
- [x] Update the bug-fix number in the branch
- [x] Update `/wiki/Releases`
- [x] Update https://github.com/CGAL/cgal/wiki/Internal-Releases
- [x] Create new tables for features/small features with closed review for the next release
- [x] Create a new issue for the check list of the next release
- [x] Update https://www.cgal.org/releases.html and change the post flags (latest-release) + remove the beta flag (latest-beta-release)
- [x] Update Travis config file of cgal-swig-bindings
- [x] Re-upload the pre-compiled demos
- [x] update PR for brew package
- [x] update PR for vcpkg
- [x] update RPM for Fedora

**Comment by afabri** [comment]
The Windows demos downloadable from the package overview are from 4.12 and not 4.13.    The uplaod of the zip files MUST be automatized.

**Comment by lrineau** [comment]
[**@afabri**](https://github.com/afabri) commented on [Jan 22, 2019, 7:23 PM GMT+1](https://github.com/CGAL/cgal/issues/3361#issuecomment-456507974 "2019-01-22T18:23:57Z - Replied by Github Reply Comments"):
> The Windows demos downloadable from the package overview are from 4.12 and not 4.13.    The uplaod of the zip files MUST be automatized.

@maxGimeno I thought it was already done... Can you check why it was not?

**Comment by maxGimeno** [comment]
I don't think any platform asks for $COLLECT_DEMOS_BINARIES in the autotest_cgal, so they are not collected. 
Keep in mind that  we don't have any active release platform on MSVC 2013 or 2015 anymore, so either we provide 2017 compiled programs either we will have to setup a specific platform for this.

**Comment by maxGimeno** [comment]
Well, every day the demo files are collected and sent to  cgaltest@cgaltest.geometryfactory.com:incoming/
under the name demos-CGAL-[release_number]_afabri-test[number]-x64_Cygwin-Windows10_MSVC2017-Release-64bits-cmake.tar.gz
After that I don't know. Maybe the cronjob that's supposed to update the files with it is down or something.

**Comment by lrineau** [comment]
There is a script in `Maintenance/public_release/scripts/precompiled_demos_zips` that packages the zip files, from the result of `demos-CGAL-[release_number]_afabri-test[number]-x64_Cygwin-Windows10_MSVC2017-Release-64bits-cmake.tar.gz`, but it has never been fully automatized in a cron job.

**Comment by lrineau** [comment]
@maxGimeno Have you already prepared pull-requests to Brew and VCpkg?

**Comment by maxGimeno** [comment]
No

**Comment by lrineau** [comment]
@sloriot: I did everything but:
> - [ ] Update `.htaccess` with the data from [here](https://gist.github.com/sloriot/4ccb76b96a3558c8177c3eda914e67a0) + test for example AABB-tree + Triangulation_2 (not sure about the last rule that seems a bit generic).

What should I do, about that?

**Comment by sloriot** [comment]
I almost made it work. If you click [here](https://doc.cgal.org/latest/Convex_hull_3/group__PkgConvexHull3Ref.html) and change the version down and then up, you'll reach the correct page. What does not work is if you go to master, even from latest which is a bit surprising.

From curl, it looks like the command `Redirect /master https://cgal.geometryfactory.com/CGAL/doc/master` is applied only after the rewrite that thus cycling.

**Comment by sloriot** [comment]
One thing I forgot to mention is the need of the `L` flag in `RewriteRule ^4.14/(.*) /latest/$1 [R=seeother,L]` to make Apache stop after the match and reevaluate the expression after replacement.

**Comment by sloriot** [comment]
master -> CGAL releases now works (redirect is not interpreted before rewrite so I simply change replaced redirect by a rewrite rule). 4.14+latest -> master works. CGAL < 4.14 -> master does not work only for the reference manual page (it would need to copy the rules from doc.cgal.org to cgal.gf.com but that does not worth it IMHO).
